### ISSUE TITLE
Models missing from Users & Permissions Application dropdown

### ISSUE THREAD

**Issue by vpoola88** [strapi/strapi#779]
**Node.js version**:
v9.8.0
**npm version**:
5.7.1
**Strapi version**:
3.0.0-alpha.11.1
**Operating system**:
macOs High Sierra 10.13.2

**Do you want to request a *feature* or report a *bug*?**
bug

**What is the current behavior?**
I've created a few models using the Content Type Builder. After creating new models and going to the Users & Permissions / Application dropdown, I'm unable to see the new models in this area. This happens on the hosted version. Locally, I'm able to see the new models.

**If the current behavior is a bug, please provide the steps to reproduce the problem**
1. Create content types.
2. Push to heroku.
3. Create new content types.
4. Go to Users & Permissions.
The models will be missing to select actions. They are visible on the left handle panel, but not within this menu.
![screen shot 2018-03-14 at 12 02 05 am](https://user-images.githubusercontent.com/1377918/37422836-6c409466-2779-11e8-89dd-66e43206814a.png)
![screen shot 2018-03-14 at 12 02 18 am](https://user-images.githubusercontent.com/1377918/37422844-70db908e-2779-11e8-9097-e591d8e70c73.png)

**What is the expected behavior?**
Can select actions on any new items created via Content Type Builder

**Comment by lauriejim** [comment]
Hi @vpoola88 !

I'm not sure at 100% but I think Heroku don't let you write files on the server.
And the content-type-builder plugin write files to create new APIs. So your need is impossible on Heroku I think.

We have to disable content-type-builder and settings-manager in production mode. Because you will not be able to give back update you will make on your Heroku instance.

**Comment by vpoola88** [comment]
Hi @lauriejim thanks for getting back!

I dont think that's the case, only b/c with further digging, I am able to see those actions through a different role. I created a "public" role, by doing so can go into the actions and see them. Any idea why its just not working for guest with that info?

Also - if I want to just use a different role with no auth, I still get a 401 unauthorized when I do a get request to those new models. If you could shed some light then on how to keep them open to any get request via any role, I can then continue tweaking just via a different role.

**Comment by lauriejim** [comment]
If you do the same actions on your local machine it works ?

Actually no idea. We have to investigate on it.

**Comment by lauriejim** [comment]
Hi @e2o, what is your Strapi version ?

**Comment by e2o** [comment]
`3.0.0-alpha.10.2`

**Comment by lauriejim** [comment]
@e2o Can you update to the last version of strapi (guides or in documentation).

**Comment by carmelyne** [comment]
Node.js version:
v9.8.0
npm version:
5.6.0
Strapi version:
3.0.0-alpha.11.1
Operating system:
macOs High Sierra 10.12.6

Experiencing the same issue with disappearing models. 

On local dev, there is such issue and I can set all the allowed actions for all the models. e.g. checking find and findone.

On production, the models do not appear under the **APPLICATION — Define all your project's allowed actions.** I can't seem to find the documentation for this settings when it's already in production mode.

**Comment by horaciomoreno10** [comment]
any update about of this?

**Comment by lauriejim** [comment]
Hey ! we will work on product stabilization for next months, we split important issues in milestones.
For this issue, we will work on it after 3.0.0-alpha.12.1

**Comment by zhuang17** [comment]
I am using the latest 3.0.0-alpha.12 version, 
i got the same issue here.

1. add a new model `blog` in `Content Type Builder` in local.
2. i can choose `blog` actions(find, findOne,...) in `Application` section of `Users & Permissions` in local.
3. when i use the same code to deploy in my prod server, however i can't see that model in `Application` section of `Users & Permissions`

I check the data of `users-permissions_permission` collection in db, we don't have `{type: "application",  controller:"blog"}` data `users-permissions_permission` in prod server.
so looks like we need some way to sync that data to server ?

**Comment by zhuang17** [comment]
and btw,  for workaround, looks like dropping the `users-permissions_permission` collection in database and restart server can work( it will reload all your models' users-permissions from your code).

(but it will remove all your previous users-permissions config, so you need to re-configure again in `Application` section of `Users & Permissions` page)

**Comment by lauriejim** [comment]
Thank you @zhuang17 for these details ! Will help to fix it.

**Comment by Aurelsicoko** [comment]
For everyone using Strapi with Heroku, there is no issue. Heroku doesn't allow to write file on the system, so the API can't be generated. It could happen that you can't see the Content-Type in the left menu or in the permissions page but if you refresh the page it will disappear. We're using a sort of optimistic response to avoid refresh but the Content-Type isn't really created and Heroku doesn't return any error.

I'm investigating for those which are not deploying on Heroku... To be continued 🙈

**Comment by Aurelsicoko** [comment]
@zhuang17 I can't reproduce the issue. I created a fresh project on my machine, then [I pushed the code on GitHub](https://github.com/Aurelsicoko/issue779), cloned and pulled the repository on my Ubuntu server and it worked as excepted.

Are you sure you're using the same database? I was using two different database so the permissions weren't matching but the problem wasn't here. I was able to see the Content-type created on my local machine after pulling the code on my Ubuntu server.

-----

@e2o @zhuang17 @carmelyne @horaciomoreno10 
We need a detailed step by step tutorial to be able to reproduce the issue. Otherwise we won't be able to fix it. Also can you confirm me that you were using the same database in development and production?

**Comment by zhuang17** [comment]
@Aurelsicoko 

Hi, sorry i didn't say it clearly.
I am using different db for dev and prod ( local Mongdb for dev, and a cloud Mongdb for the prod).

**Comment by Aurelsicoko** [comment]
Okay, so it makes sense that the permissions are not the same, etc. We're going to offer a way to migrate configuration data from an environment to another, see https://github.com/strapi/strapi/issues/672#issuecomment-381662918.

Can you confirm me that there is no issue so I'll be able to close this one?

**Comment by carmelyne** [comment]
@Aurelsicoko I am using the same type of db (MongoDB) for both dev and production (self hosted MongoDB on ubuntu). Also, still experiencing the same problem.

I would normally give you a step by step but I just can't recreate it. 

Although, if I create a new role, I can see all the collections again, and, for example, check find and findone just fine.

No matter what I do, even running dev mode on my production server, the collections under the Application toggle is limited to 3. The first three collections I had on my initial set up. Hope that sheds some light.

**Comment by carmelyne** [comment]
Side note: Dropping **users-permissions_permission** and _restarting the server_ resolved the issue. Thanks for that info @zhuang17.

**Comment by Aurelsicoko** [comment]
@carmelyne I'm almost sure it was related to duplicate entries in the `users-permissions_permission` table. I'm closing the issue but I won't lock the conversation. Feel free to ping me if the issue re-appears again.

**Comment by bruce0205** [comment]
@Aurelsicoko same issue, still missing..

I am using different databases (one for dev, the other for prod) on the same mongo server (localhost).

Step 1:  I ran「npm start」, then using "Content Type Builder" to create model.
Step 2: I ran「NODE_ENV=production npm start」, but I cannot set any permissions for that model on the "Roles & Permissions" of menu.

Node.js version:
v9.9.0
npm version:
5.6.0
Strapi version:
3.0.0-alpha.12.7.1
Operating system:
macOs High Sierra 10.13.5
